share/doc/wake/logging.md

```diff
@@ -0,0 +1,152 @@
+|--no-tty<sup>*</sup> |          |         |    x    |     x     |      x     |    x     |         |        |
+|(default)            |          |         |         |     x     |      x     |    x     |         |        |
+
+<sup>*</sup> `--no-tty` also prevents any coloring from being applied.
+
+Unless overridden with `--stderr=...` the *only* thing that ever appears on wake's stderr is outputs from loggers
+with log level "error" (e.g. `logError`).
+
+The additional command line arguments `--fd:3`, `--fd:4`, `--fd:5` can be used to output to file descriptors 3, 4, 5.
+One example shell command would be, for some wake code that had `mkLogLevel "foo"`:
+
+```
+wake --fd:3="foo" mywakecode 3>foo.txt
+```
+
+These can be concatenated with commas and include the normal log levels, so to capture both `mkLogLevel "foo"` and `logInfo` to a file:
+
+```
+wake --fd:3="foo,info" mywakecode 3>foo_and_info.txt
+```
+
+## What Gets Sent To Loggers
+
+### Job Execution
+
+Every `Job` run by wake tracks its own stdout and stderr,
+and when the underlying process prints to stdout or stderr,
+it will go to the `Job`-specific stdout and stderr,
+not the stdout and stderr of the `wake` process itself.
+
+When writing your wake code,
+you can control which loggers these `Job`-specific stdout and stderr are sent to.
+This will influence whether they appear on the `wake` process's own stdout,
+stderr, or other arbitrary file descriptors.
+These can be controlled with `setPlanStdout`, `setPlanStderr` before you `runJob`.
+These default to `logInfo` and `logError` respectively.
+
+Regardless of to which logger you direct the `Job`'s `Stderr` and `Stdout`,
+`wake` will store all the text in its database,
+discarding the information about the logger it was destined for at the time it was stored.
+It will however know whether it came from the `Job`'s stderr/stdout.
+
+### Other Things that Appear on Wake's Output Streams During Execution
+
+In addition to stdout and stderr, each executing `Job` will send the command
+it is executing to a logger. The logger it is sent to by defalt is `logEcho`,
+and this can be overridden with `setPlanEcho`.
+
+The results of calling `print`/`println` within `wake` code are sent to `logReport` by default.
+To send them somewhere else, use `printLevel`, `printlnLevel` instead.
+
+The final output of whatever wake function or expression is called is always directed to
+wake's `stdout` unless `--quiet` is set.
+This is outside of the logger mechanism,
+so it is not possible to redirect this within wake code or with command line flags.
+
+Similarly, the progress bar is always sent to wake's `stdout` unless `--no-tty`
+or `--quiet` is set.
+
+## What is Displayed to user on Wake Database Commands
+
+The `--verbse` and `--metadata` flags impact what is shown per-job by the database
+(`--last, --failed, --job`, etc) commands.
+They do not impact which `Job`s information is shown for.
```

Comment:
Specifically, this is talking about `--verbose` when it's given alongside a database command.  `--verbose` *without* a database command is described in the table above.  Maybe put the predicate first?
```suggestion
When used with a database command (`--last, --failed, --job`, etc),
the `--verbose` and `--metadata` flags impact what is shown per-job by the database.
They do not impact the selection of the `Job`s for which that information will be shown.
```